linera-chain/src/manager.rs

```diff
@@ -549,6 +550,21 @@ impl ChainManager {
     fn is_super(&self, owner: &Owner) -> bool {
         self.ownership.super_owners.contains_key(owner)
     }
+
+    fn update_locked(&mut self, new_locked: Certificate) {
+        self.locked = Some(new_locked);
+        self.locked_blobs = self.proposed_blobs.clone();
```

Comment:
IMHO this indicates that we have a "flow problem" - the `update_locked` is "promoting" the `self.proposed_blobs` to `self.locked_blobs` but it isn't doing anything with `self.proposed`. IMHO this whole struct is asking for a refactor but I don't have good ideas for it yet.

Reply:
And what if the new locked block isn't the same as the previous proposed block?